java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/ScopedAccessWithHostBuffer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.cuvs.internal;
+
+import com.nvidia.cuvs.CuVSResources;
+import java.lang.foreign.MemorySegment;
+
+class ScopedAccessWithHostBuffer implements CuVSResources.ScopedAccess {
```

Comment:
Should we consider renaming this class to `ScopedAccessWithPinnedMemoryBuffer`?

Reply:
Hmm.  I see that the `PinnedMemoryBuffer` used to construct this is actually part of `CuvsResourcesImpl`.  And this class is only used with pinned-memory.

I'm trying to work out if there's a different way of arranging this.